packages/compartment-mapper/src/types/external.d.ts

```diff
@@ -0,0 +1,352 @@
+  ModuleTransformsOption &
+  ImportingOptions &
+  ExitModuleImportHookOption &
+  LinkingOptions;
+
+export type SyncArchiveLiteOptions = SyncOrAsyncArchiveOptions &
+  SyncModuleTransformsOption &
+  SyncImportingOptions &
+  ExitModuleImportNowHookOption;
+
+export type ArchiveOptions = Omit<MapNodeModulesOptions, 'language'> &
+  ArchiveLiteOptions;
+
+export type SyncArchiveOptions = Omit<MapNodeModulesOptions, 'languages'> &
+  SyncArchiveLiteOptions;
+
+/**
+ * Options for `loadLocation()`
+ */
+export type LoadLocationOptions = ArchiveOptions & SyncArchiveOptions;
+
+/**
+ * Options for `importLocation()` necessary (but not sufficient--see
+ * {@link ReadNowPowers}) for dynamic require support
+ */
+export type SyncImportLocationOptions = SyncArchiveOptions & ExecuteOptions;
+
+export type ImportLocationOptions = ArchiveOptions & ExecuteOptions;
+
+// ////////////////////////////////////////////////////////////////////////////////
+// Single Options
+
+type SearchSuffixesOption = {
+  /**
+   * Suffixes to search if the unmodified specifier is not found. Pass `[]` to
+   * emulate Node.js' strict behavior. The default handles Node.js' CommonJS
+   * behavior.
+   * Unlike Node.js, the Compartment Mapper lifts CommonJS up, more like a
+   * bundler, and does not attempt to vary the behavior of resolution depending
+   * on the language of the importing module.
+   */
+  searchSuffixes?: string[];
+};
+
+type SourceMapHookOption = {
+  sourceMapHook?: SourceMapHook;
+};
+
+type ModulesOption = {
+  modules?: Record<string, any>;
+};
+
+type ExitModuleImportHookOption = {
+  importHook?: ExitModuleImportHook;
+};
+
+type ExitModuleImportNowHookOption = {
+  importNowHook?: ExitModuleImportNowHook;
+};
+
+type ParserForLanguageOption = {
+  parserForLanguage?: ParserForLanguage;
+};
+
+type CompartmentOption = {
+  Compartment?: typeof Compartment;
+};
+
+export type ModuleTransformsOption = {
+  moduleTransforms?: ModuleTransforms;
+};
+
+export type SyncModuleTransformsOption = {
+  syncModuleTransforms?: SyncModuleTransforms;
+};
+
+export type PolicyOption = {
+  policy?: any;
+};
+
+type LanguageForExtensionOption = {
+  languageForExtension?: LanguageForExtension;
+};
+
+// ////////////////////////////////////////////////////////////////////////////////
+// Common option groups:
+
+export type SyncOrAsyncArchiveOptions = Partial<{
+  captureSourceLocation: CaptureSourceLocationHook;
+}> &
+  ParserForLanguageOption &
+  CompartmentOption &
+  PolicyOption;
+
+type SyncOrAsyncImportingOptions = SearchSuffixesOption & SourceMapHookOption;
+
+type ImportingOptions = ModulesOption &
+  SyncOrAsyncImportingOptions &
+  ExitModuleImportHookOption;
+
+type SyncImportingOptions = ModulesOption &
+  SyncOrAsyncImportingOptions &
+  ExitModuleImportNowHookOption;
+
+type LinkingOptions = ParserForLanguageOption &
+  CompartmentOption &
+  LanguageForExtensionOption &
+  SyncModuleTransformsOption &
+  ModuleTransformsOption;
```

Comment:
Much of what I want to achieve here is sensible groupings of options here, and to ensure that all of these options get threaded to their constituent subroutines. For example, `languageForExtension` should be threaded from `loadLocation`, `makeArchive`, and such all the way down to `mapNodeModules` where it’s used.